server/modules/dm-packer/DmPacker.js

```diff
@@ -80,18 +80,22 @@ module.exports = function DmPacker(logger) {
 
   this.uploadCodeDeployPackage = function (destination, stream, s3client) {
     return new Promise((resolve, reject) => {
-      stream.on('error', err => reject(err));
-      s3client.upload({
-        Bucket: destination.bucket,
-        Key: destination.key,
-        Body: stream
-      }, (err, data) => {
-        if (err) {
-          reject(err);
-        } else {
-          resolve(data);
-        }
-      });
+      try {
```

Comment:
this part looks ugly, why not use promises chaining? it would get reduced to 5 lines